### PR TITLE
DOC: EX01 ({Categorical, Interval, Multi, Datetime, Timedelta}-Index)

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [x] Towards #37875.

Also removing some items from `ci/code_checks.sh` that have already had examples on the main branch.

### CODE REVIEW

**Comment by DeaMariaLeon** [comment]
Thank you for your PR @Charlie-XIAO!  

Just a comment...
Some of items that you removed from the file `ci/code_checks.sh` thinking that they already have examples, should not be removed. Many methods share one (the same) docstring and it _looks_ like that.

For example `pandas.CategoricalIndex.codes` seems on the dev website as it had an example already. But the example is actually for `pandas.Categorical.codes`.  

I only checked a couple of methods that were removed. It would be great if you could double check the rest please. Thanks.

**Comment by DeaMariaLeon** [comment]
Thank you for your PR @Charlie-XIAO!

Some of items that you removed from the file `ci/code_checks.sh` thinking that they already have examples, should not be removed. Many methods share one (the same) docstring and it _looks_ like that.

For example `pandas.CategoricalIndex.codes` seems on the dev website as it had an example already. But the example is actually for `pandas.Categorical.codes`.  

I only checked a couple of methods that were removed. It would be great if you could double check the rest please. Thanks a lot, your help is very much appreciated. :-)

**Comment by Charlie-XIAO** [comment]
@DeaMariaLeon Thanks for the reminder! I see what you mean.

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -760,23 +760,32 @@ def categories(self) -> Index:
 
         Examples
         --------
+        For :class:`pandas.Series`:
 
-        For Series:
```
Comment (MarcoGorelli) [inline pandas/core/arrays/categorical.py:764]: nice!

File: pandas/core/indexes/category.py [hunk 2]
```diff
@@ -280,8 +280,43 @@ def equals(self, other: object) -> bool:
         Returns
         -------
         bool
-            If two CategoricalIndex objects have equal elements True,
-            otherwise False.
+            ``True`` if two :class:`pandas.CategoricalIndex` objects have equal
+            elements, ``False`` otherwise.
+
+        Examples
+        --------
```
Comment (MarcoGorelli) [inline pandas/core/indexes/category.py:287]: great example here!

File: pandas/core/arrays/categorical.py [hunk 3]
```diff
@@ -760,23 +760,32 @@ def categories(self) -> Index:
...
-
-        >>> ser = pd.Series(["a", "b", "c", "a"], dtype="category")
+        >>> ser = pd.Series(['a', 'b', 'c', 'a'], dtype='category')
         >>> ser.cat.categories
         Index(['a', 'b', 'c'], dtype='object')
 
-        >>> raw_cat = pd.Categorical(["a", "b", "c", "a"], categories=["b", "c", "d"],)
+        >>> raw_cat = pd.Categorical(['a', 'b', 'c', 'a'], categories=['b', 'c', 'd'])
```
Comment (MarcoGorelli) [inline pandas/core/arrays/categorical.py:769]: just for the future - in general, let's not bother with stylistic changes like changing double quotes to single quotes, at least until we have an automated tool to enforce this on docstrings

**Review by MarcoGorelli — APPROVED** [review]
nice, thanks for your work here! and thanks Dea for reviewing

**Pull request merged**